AfterClass-io/afterclass.io-v2
# feat(ui): review item responsive mobile viewport

closes #204 

## Changes

- add responsive variant size on `reviewItemTheme`, `reviewSectionTheme` and `profileTheme`
- collapse reviewmodal into a subcomponent of review item
- collapse ReviewHeader into 2 subcomponent, `RevieweeGroup` and `ReviwerGroup`
- abstract `Profile` component from `ProfileSchool` and `ProfileReviewer`

## How to Test

1. head to home page
2. open devtools
3. make viewport responsive
4. check modal changes when viewport changes

## Preview / Screenshots

### `md` viewport

![image](https://github.com/user-attachments/assets/9e59b3f8-2e63-4e79-b5ba-9c26921458b7)

### `sm` viewport

![image](https://github.com/user-attachments/assets/0eb8c8d4-de6a-4c73-8ff7-5f6e3cb288ae)

## Checklist
